src/platform/intel/cavs/platform.c

```diff
@@ -67,7 +67,10 @@ static const struct sof_ipc_fw_ready ready
 		.abi_version = SOF_ABI_VERSION,
 		.src_hash = SOF_SRC_HASH,
 	},
-#if CONFIG_CAVS_IMR_D3_PERSISTENT
+#if CONFIG_CAVS_IMR_D3_PERSISTENT && CONFIG_ADSP_IMR_CONTEXT_SAVE
+	.flags = DEBUG_SET_FW_READY_FLAGS | SOF_IPC_INFO_D3_PERSISTENT |
+		SOF_IPC_INFO_CONTEXT_SAVE,
+#elif CONFIG_CAVS_IMR_D3_PERSISTENT
```

Comment:
We are not reading the IPC3 `sof_ipc_fw_ready` message from mailbox in case of IPC4.
Reference firmware do not passes any information via mailbox for FW_READY, iow IPC4 does not define payload for FW_READY.

Reply:
The IPC4 way of fw configuration query is via LAGE_CONFIG_GET:FW_CONFIG message. It has definition for various firmware parameters as tlv tuple bundle.
The reference IPC4 FW_READY notification does not have payload and I think if we want to keep the IPC4 name we should not introduce incompatible changes to the basic and common messages, notifications.

Reply:
Yes, I agree. An IPC3 (SOF specific) message in the mailbox as payload for IPC4  FW_READY notification is not going to be backwards compatible with the reference, it is not going to be compatible at all since the reference is _not_ placing anything in the mailbox for FW_READY, especially it is not going to place an IPC3 message in there which is part of the IPC3 protocol - in a similar way that IPC4 FW_READY does not have payload.

Reply:
@ujfalusi why we can't development IPC4 without breaking windows driver ? And if we are stick to current IPC4, we can't use LAGE_CONFIG_GET:FW_CONFIG  since it doesn't support this query and we can't add more config, right ?

Reply:
@RanderWang I think direction now is to take this flag as a manifest flag. We should make it clear and add a comment to the FW ready message that the fields in FW ready should only be for runtime generated information (i.e. build time information would go in the manifest). Thanks !